### PR TITLE
[LOOP-1431] reset basal delivery state when pump manager is deleted

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1431

when pump manager is deleted, reset the basalDeliveryState

### CODE REVIEW

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -251,11 +251,14 @@ final class StatusTableViewController: LoopChartsTableViewController {
     }
 
     private func registerPumpManager() {
-        if let pumpManager = deviceManager.pumpManager {
-            self.basalDeliveryState = pumpManager.status.basalDeliveryState
-            pumpManager.removeStatusObserver(self)
-            pumpManager.addStatusObserver(self, queue: .main)
+        guard let pumpManager = deviceManager.pumpManager else {
```
Comment (ps2) [inline Loop/View Controllers/StatusTableViewController.swift]: [nit] The guard could be removed, and the pump manager accessed with optional chaining below.

**Review by ps2 — APPROVED** [review]
LGTM!

**Review by darinkrauss — APPROVED** [review]
LGTM! ⚙️

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -251,11 +251,10 @@ final class StatusTableViewController: LoopChartsTableViewController {
     }
 
     private func registerPumpManager() {
-        if let pumpManager = deviceManager.pumpManager {
-            self.basalDeliveryState = pumpManager.status.basalDeliveryState
-            pumpManager.removeStatusObserver(self)
-            pumpManager.addStatusObserver(self, queue: .main)
-        }
+        self.basalDeliveryState = deviceManager.pumpManager?.status.basalDeliveryState
+        self.bolusState = deviceManager.pumpManager?.status.bolusState ?? .none
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift]: Shouldn't the `bolusState` also be an optional set to nil when there is no pod? It is a different state than `.none` (which means there is a pod, but no current ongoing bolus happening).

**Review by darinkrauss — APPROVED** [review]
One concern about the inconsistency between `basalDeliveryState` and `bolusState` when there is no pod and when there is no pump manager.

**Review by darinkrauss — APPROVED** [review]
LGTM! 🚀

**Pull request merged**